[docs] add explanation to `release_memory()` (#34911)
huggingface/transformers: docs/source/en/llm_tutorial_optimization.md

```python
flush()
```

> In the recent version of the accelerate library, you can also use a device-agnostic utility method called `release_memory()`, which takes various hardware backends like XPU, MLU, NPU, MPS etc. into account:

Reviewer: Would also be nice to add a link to `release_memory` for more curious users

```suggestion
From the Accelerate library, you can also use a device-agnostic utility method called [release_memory](https://github.com/huggingface/accelerate/blob/29be4788629b772a3b722076e433b5b3b5c85da3/src/accelerate/utils/memory.py#L63), which takes various hardware backends like XPU, MLU, NPU, MPS, and more into account.
```
